src/compiler/moduleNameResolver.ts

```diff
@@ -339,7 +339,13 @@ namespace ts {
         if (resolved) {
             const { fileName, packageId } = resolved;
             const resolvedFileName = options.preserveSymlinks ? fileName : realPath(fileName, host, traceEnabled);
-            resolvedTypeReferenceDirective = { primary, resolvedFileName, packageId, isExternalLibraryImport: pathContainsNodeModules(fileName) };
+            resolvedTypeReferenceDirective = {
+                primary,
+                resolvedFileName,
+                originalFileName: fileName === resolvedFileName ? undefined : fileName,

```

Comment:
I think this should be `originalPath` just like in `ResolvedModuleFull`.. 
checkout createResolvedModuleWithFailedLookupLocations , you would want to do something similar

Reply:
It is string and not Path, check in `nodeModuleNameResolverWorker`  though it seems like it should compare paths there and not strings as there could be case difference.( but thats separate issue that @amcasey might want to look in for his realpath fix he was working on )

Reply:
I think if it’s not going to be normalized into a `Path`, the property name should use `fileName` and not `path`. And if we don’t preserve the file name, I think test cases like https://github.com/microsoft/TypeScript/blob/master/tests/cases/fourslash/autoImportSymlinkCaseSensitive.ts will start having issues.

Reply:
Sorry i wasnt very clear in my comment.. What i was saying was that comparison needs to be between `Path` but value needs to be string.

Reply:
Ah, I see, just for the equality comparison. It doesn’t look like I have easy access to the current directory or `getCanonicalFileName` (the latter is optional on `ModuleResolutionHost`) and this is public API. What do you suggest doing?

Reply:
I dont think you need to fix it in this PR. But module resolution host has optional `getCurrentDirectory` so we can add `useCaseSensitiveFileNames?(): boolean`  and use Path if both methods exist and compare filenames otherwise. 
you would also want to make sure our all internal `moduleResolutionHosts` have these implemented. Again dont think this is part of your PR but needs to be handled at some point for @amcasey realpath native methods to be used